src/ui/index.js

```diff
@@ -77,12 +87,74 @@ export default async (kbnServer, server, config) => {
       },
       vars: defaults(app.getInjectedVars() || {}, uiExports.defaultInjectedVars),
     };
-
+    if (kibanaTranslations.length <= 0) {
+      const language = await getTranslationLanguage(acceptLanguages);
+      kibanaTranslations = await i18nPlugin.getRegisteredLanguageTranslations(language);
+    }
     return this.view(app.templateName, {
       app: app,
       loadingGif: loadingGif,
       kibanaPayload: payload,
       bundlePath: `${config.get('server.basePath')}/bundles`,
+      welcomeMessage: translate('CORE-WELCOME_MESSAGE'),
+      welcomeError: translate('CORE-WELCOME_ERROR'),
     });
   });
 };
+
+function translate(key) {
+  if (!(key in kibanaTranslations)) {
+    return null;
+  }
+  return kibanaTranslations[key];
+}
+
+async function getTranslationLanguage(acceptLanguages) {
+  let langStr = '';
+  let foundLang = false;
+
+  if (acceptLanguages === null || acceptLanguages.length <= 0) {
+    return DEFAULT_LANGUAGE;
+  }
+
+  const acceptLangsLen = acceptLanguages.length;
+  const registeredLanguages = await i18nPlugin.getRegisteredTranslationLanguages();
+
+  for (let indx = 0; indx < acceptLangsLen; indx++) {
+    const language = acceptLanguages[indx];
+    if (language.region) {
```

Comment:
Is this more to do with multi-part region codes, i.e. 'az-AZ-Cyrl' will be treated as 'az-AZ'? It looks like the region would be returned correctly for code of `zn` and region `Hans` or `Hant` as it is not mutli-part region.

Maybe as you suggest we use the [negotiator](https://github.com/jshttp/negotiator#accept-language-negotiation) instead of [language parser](https://github.com/opentable/accept-language-parser)?